server/stream.go

```diff
@@ -1206,43 +1243,65 @@ func (s *Server) checkStreamCfg(config *StreamConfig, acc *Account) (StreamConfi
 
 		}
 	}
-	if len(cfg.Sources) > 0 {
-		// check for duplicates
-		var iNames = make(map[string]struct{})
-		for _, src := range cfg.Sources {
-			if !isValidName(src.Name) {
-				return StreamConfig{}, NewJSSourceInvalidStreamNameError()
-			}
-			if _, ok := iNames[src.composeIName()]; !ok {
-				iNames[src.composeIName()] = struct{}{}
-			} else {
-				return StreamConfig{}, NewJSSourceDuplicateDetectedError()
-			}
+
+	// check for duplicates
+	var iNames = make(map[string]struct{})
+	for _, src := range cfg.Sources {
+		if !isValidName(src.Name) {
+			return StreamConfig{}, NewJSSourceInvalidStreamNameError()
 		}
-		for _, src := range cfg.Sources {
-			// Do not perform checks if External is provided, as it could lead to
-			// checking against itself (if sourced stream name is the same on different JetStream)
-			if src.External == nil {
-				exists, maxMsgSize, subs := hasStream(src.Name)
-				if len(subs) > 0 {
-					streamSubs = append(streamSubs, subs...)
-				}
-				if exists {
-					if cfg.MaxMsgSize > 0 && maxMsgSize > 0 && cfg.MaxMsgSize < maxMsgSize {
-						return StreamConfig{}, NewJSSourceMaxMessageSizeTooBigError()
-					}
+		if _, ok := iNames[src.composeIName()]; !ok {
+			iNames[src.composeIName()] = struct{}{}
+		} else {
+			return StreamConfig{}, NewJSSourceDuplicateDetectedError()
+		}
+		// Do not perform checks if External is provided, as it could lead to
+		// checking against itself (if sourced stream name is the same on different JetStream)
+		if src.External == nil {
+			exists, maxMsgSize, subs := hasStream(src.Name)
+			if len(subs) > 0 {
+				streamSubs = append(streamSubs, subs...)
+			}
+			if exists {
+				if cfg.MaxMsgSize > 0 && maxMsgSize > 0 && cfg.MaxMsgSize < maxMsgSize {
+					return StreamConfig{}, NewJSSourceMaxMessageSizeTooBigError()
 				}
+			}
 
-				continue
+			if (src.FilterSubject != _EMPTY_ || src.SubjectTransformDest != _EMPTY_) && len(src.SubjectTransforms) != 0 {
+				return StreamConfig{}, NewJSSourceMultipleFiltersNotAllowedError()
 			}
-			if src.External.DeliverPrefix != _EMPTY_ {
-				deliveryPrefixes = append(deliveryPrefixes, src.External.DeliverPrefix)
+
+			if len(src.SubjectTransforms) > 0 {
```

Comment:
I think we can drop this check.
range loop will not execute anyway If `len() == 0`

Reply:
Correct, I missed cleaning that one, thanks.